src/main.cpp

```diff
@@ -2472,7 +2472,8 @@ int32_t ComputeBlockVersion(const CBlockIndex* pindexPrev, const Consensus::Para
 
     for (int i = 0; i < (int)Consensus::MAX_VERSION_BITS_DEPLOYMENTS; i++) {
```

Comment:
I'll just re-ask this question here since I'm not sure it was in the right line on the last PR. This loop is looping through all the version bits defined in consensus/params.h and then the IsVersionBitRejected function loops through each of the values in the config to check if something is rejected. Are there any of these which we don't want people to be able to disable;

https://github.com/aguycalled/navcoin-core/blob/5a0aa5493388076e50b5c671c3f9dd80618e3aef/src/consensus/params.h#L16

I assume if someone turns off the segwit bit for example, it will have no effect. But it would be good to double check with you guys.

Reply:
As far as I understand, this won't affect any of the bit versions that have already been accepted and activated,

@aguycalled ^

Reply:
it's more of a formal aspect. you can change the logic to something like
```
if (state == THRESHOLD_LOCKED_IN || state == THRESHOLD_ACTIVE || (state == THRESHOLD_STARTED
                 && !IsVersionBitRejected(params, (Consensus::DeploymentPos)i))) {
```
so IsVersionBitRejected() only applies on the STARTED state.
with the logic proposed by @red010b37 the bits would also be affected in the other states, (without any practical consequence as the necessary bit masks are set after this loop).
for sanity i consider IsVersionBitRejected should only affect when on the STARTED state

Reply:
@aguycalled that makes sense, I will make the change suggested

Reply:
@craigmacgregor @aguycalled I have updated the code to reflect the conversation above